### ISSUE TITLE
Enhance editor sidebar title UI

### ISSUE THREAD

**Issue by t-hamano** [WordPress/create-block-theme#604]
The editor sidebar title text is confusing because it's included in the back button. Additionally, screen readers read the title text for this button, so they cannot tell you that it is a "back" button.

![image](https://github.com/WordPress/create-block-theme/assets/54422211/b2bae806-0278-4b49-af99-0a96882571af)

I suggest separating the button and title text.

![image](https://github.com/WordPress/create-block-theme/assets/54422211/a6121f2c-5b0c-44f8-9882-0dc9f65826d4)

This should also match the Global Styles UI.

![image](https://github.com/WordPress/create-block-theme/assets/54422211/07025e0e-37cd-4cb2-9b0a-d0654f7befa0)
